core/client/fs/src/main/java/alluxio/client/file/BaseFileSystem.java

```diff
@@ -135,11 +126,8 @@ public synchronized void close() throws IOException {
     // TODO(zac) Determine the behavior when closing the context during operations.
     if (!mClosed) {
       mClosed = true;
-      if (mCachingEnabled) {
```

Comment:
put this logic back to `FileSystem` interface so all classes implementing the same interface can benefit from the same logic.